src/initdt.jl

```diff
@@ -250,9 +250,9 @@ end
         @warn("First function call produced NaNs. Exiting. Double check that none of the initial conditions, parameters, or timespan values are NaN.")
     end
 
-    if Base.promote_op(/, typeof(u0), typeof(oneunit(t))) !== typeof(f₀)
-        throw(TypeNotConstantError(Base.promote_op(/, typeof(u0), typeof(oneunit(t))),
-            typeof(f₀)))
+    inferredtype = Base.promote_op(/, typeof(u0), typeof(oneunit(t)))
+    if inferredtype !== Any && inferredtype !== typeof(f₀)
+        throw(TypeNotConstantError(inferredtype, typeof(f₀)))
```

Comment:
This has been an error before this PR. You'd like us to switch it to a warning instead?

Reply:
It should be an error, it's just catching the error earlier so the user doesn't get a very hard stack trace

Reply:
Then no change needed.

Reply:
makes sense. actually, the check we want here is that `f0 isa inferred_type` since that will do the right thing in the case where inference is slightly but not totally imprecise